crates/factor-outbound-http/src/wasi.rs

```diff
@@ -317,19 +350,21 @@ async fn send_request_handler(
         (sender, worker)
     };
 
-    // at this point, the request contains the scheme and the authority, but
-    // the http packet should only include those if addressing a proxy, so
-    // remove them here, since SendRequest::send_request does not do it for us
-    *request.uri_mut() = http::Uri::builder()
-        .path_and_query(
-            request
-                .uri()
-                .path_and_query()
-                .map(|p| p.as_str())
-                .unwrap_or("/"),
-        )
-        .build()
-        .expect("comes from valid request");
+    if !is_http2 {
+        // at this point, the request contains the scheme and the authority, but
+        // the http packet should only include those if addressing a proxy, so
+        // remove them here, since SendRequest::send_request does not do it for us
```

Comment:
This is my own ignorance but if we change the URI to remove the authority then how does it know where to send the request?